packages/vite-plugin-svelte/src/utils/options.ts

```diff
@@ -342,10 +360,30 @@ export function buildExtraViteConfig(
 				plugins: [{ name: facadeEsbuildSveltePluginName, setup: () => {} }]
 			}
 		};
+	} else {
+		if (options.disableDependencyReinclusion !== true) {
+			const disabledReinclusions = options.disableDependencyReinclusion || [];
+			if (disabledReinclusions.length > 0) {
+				log.debug(`not reincluding transitive dependencies of`, disabledReinclusions);
+			}
+			const transitiveDepsToInclude = depsConfig.optimizeDeps.include.filter((dep) =>
+				disabledReinclusions.every((disabled) => !dep.includes(disabled))
```

Comment:
What's tested here exactly? The every not could be a some instead and why string.includes instead of equal s?

Reply:
If the user disable reinclusions for some packages, we filter those out from `depsConfig.optimizeDeps.include`. It's using `include` since the specifiers could have `>` for deep optimize.

Do you mean it could be this? 
```suggestion
				!disabledReinclusions.some((disabled) => dep.includes(disabled))
```

Reply:
yes for every vs some, but the `dep.includes` part is dangerous as it could hit false positives when one dependency name is completely contained in another one, eg `svelte-foo` and `not-svelte-foo-but-close`

Reply:
ah yeah i didn't thought about that. good call, we should probably fix that